typing/typecore.ml

```diff
@@ -5532,6 +5541,36 @@ let report_unification_error ~loc ?sub env err
       ?type_expected_explanation txt1 txt2
   ) ()
 
+let report_this_function ppf funct =
+  if Typedtree.exp_is_short funct then
+    let pexp = Untypeast.untype_expression funct in
+    Format.fprintf ppf "The function '%a'" Pprintast.expression pexp
+  else Format.fprintf ppf "This function"
+
+let report_too_many_arg_error ~funct ~func_ty ~previous_arg_loc
+    ~extra_arg_loc loc =
+  let open Location in
+  let app_loc =
+    (* Span the application, including the extra argument. *)
+    { loc_start = loc.loc_start;
+      loc_end = extra_arg_loc.loc_end;
+      loc_ghost = false }
+  and tail_loc =
+    (* Possible location for a ';'. *)
+    let loc_start = previous_arg_loc.loc_end in
+    { loc_start;
+      loc_end = { loc_start with pos_cnum = loc_start.pos_cnum + 1 };
+      loc_ghost = false }
+  in
```

Comment:
There is a corner case here: there is not necessarily a white space after the last valid argument. For instance
```ocaml
let f x y z = x + y + z
let t = f 0 1
    (+)2
```
yields
```
File "test.ml", line 3, characters 7-8:
3 |     (+)2
           ^
  Hint: Did you forget a ';'?
File "test.ml", line 3, characters 7-8:
3 |     (+)2
           ^
  This argument is not expected.
```
One option that might work would be to widen the highlighting interval to include the last character of the previous arg:
```
3 |     (+)2
          ^^
  Hint: Did you forget a ';'?
  ```
  to make it clearer that we are proposing to insert `;` in the middle of the interval.

Reply:
Good idea. It's nice when there's a space too, it shows that the semicolon has to be inserted "after the argument".